### PR TITLE
Fold the map/try_map APIs into get/try_get

### PR DESCRIPTION
**Pull request by Georgiy-Tugai**

### CODE REVIEW

File: flecs_ecs/examples/flecs/entities/entity_hierarchy.rs [hunk 1]
```diff
@@ -23,7 +23,7 @@ fn iterate_tree(entity: EntityView, position_parent: &Position) {
     println!("{} [{:?}]", entity.path().unwrap(), entity.archetype());
 
     // Map is the same as get, but allows you to return a value
```
Comment (Indra-db) [inline flecs_ecs/examples/flecs/entities/entity_hierarchy.rs:25]: update comment

File: flecs_ecs/src/core/entity_view/entity_view_const.rs [hunk 2]
```diff
@@ -715,7 +717,9 @@ impl<'a> EntityView<'a> {
     ///
     /// # Returns
     ///
-    /// - If the callback has ran.
+    /// - If the callback returns `Option`, acts like [`and_then`](Option::and_then); returns the `Option` if the callback was run, `None` otherwise.
+    ///   If you need to differentiate between "None returned from callback" and "None because the callback was skipped", use the `Option` wrapper
```
Comment (Indra-db) [inline flecs_ecs/src/core/entity_view/entity_view_const.rs:721]: I'm having trouble understanding this, I request a doc test example :D

File: flecs_ecs/src/core/utility/traits/option_or_zst.rs [hunk 3]
```diff
@@ -0,0 +1,33 @@
+/// Utility trait for `try_get` callbacks
+///
+/// If the callback returns `Option`, pass it through.
+/// If the callback returns `()`, convert the `try_get` status into a `bool` instead.
+pub trait OptionOrZST {
```
Comment (Indra-db) [inline flecs_ecs/src/core/utility/traits/option_or_zst.rs:5]: I know naming is difficult, but any chance we can come up with a better name? initially when I was looking at try_get, I thought this trait meant "you can only `get` options or zst's".

File: flecs_ecs/src/core/world.rs [hunk 4]
```diff
@@ -1852,17 +1848,46 @@ impl World {
...
     where
         T::OnlyType: ComponentOrPairId,
+        Return: OptionOrZST,
     {
         let entity = EntityView::new_from(
             self,
             <<T::OnlyType as ComponentOrPairId>::CastType>::id(self),
         );
-        entity.get::<T>(callback);
+        entity.try_get::<T>(callback)
```
Comment (Indra-db) [inline flecs_ecs/src/core/world.rs:1872]: why did this become try_get from get?

**Review by Indra-db — CHANGES_REQUESTED** [review]
make sure there's no `map` references in docs, comments anywhere before finishing up the work. 

Tests seem to fail as well, so best to check that out.

File: flecs_ecs/src/core/world.rs (hunk 4 above)
Comment (Georgiy-Tugai) [reply flecs_ecs/src/core/world.rs:1872]: It's an artefact of the diff algorithm used by github; it's getting confused by the code going from "try_get followed by get" to "try_get, get, try_get, get" due to the trait and impl

File: flecs_ecs/src/core/utility/traits/option_or_zst.rs (hunk 3 above)
Comment (Georgiy-Tugai) [reply flecs_ecs/src/core/utility/traits/option_or_zst.rs]: Resolved by refactoring so the trait isn't needed anymore

File: flecs_ecs/src/core/entity_view/entity_view_const.rs (hunk 2 above)
Comment (Georgiy-Tugai) [reply flecs_ecs/src/core/entity_view/entity_view_const.rs]: Resolved by refactoring (and examples)

**Review by Indra-db — APPROVED** [review]
LGTM!

**Pull request merged**